src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,379 @@
+     */
+    private val cakeContainerPattern by patternGroup.pattern(
+        "cake.container",
+        "Ender Chest \\(\\d{1,2}/\\d{1,2}\\)|.*Backpack(?:§r)? \\(Slot #\\d{1,2}\\)|New Year Cake Bag",
+    )
+
+    /**
+     * REGEX-TEST: New Year Cake Bag
+     */
+    private val cakeBagPattern by patternGroup.pattern(
+        "cake.bag",
+        "New Year Cake Bag",
+    )
+
+    /**
+     * REGEX-TEST: Auctions Browser
+     * REGEX-TEST: Auctions: "Test"
+     */
+    private val auctionBrowserPattern by patternGroup.pattern(
+        "auction.search",
+        "Auctions Browser|Auctions: \".*",
+    )
+
+    /**
+     * REGEX-TEST: Auctions: "New Year C
+     */
+    private val auctionCakeSearchPattern by patternGroup.pattern(
+        "auction.cakesearch",
+        "^Auctions: \"New Year C.*$",
+    )
+
+    private val tracker = SkyHanniTracker("New Year Cake Tracker", { Data() }, { it.cakeTracker }) {
+        drawDisplay(it)
+    }
+
+    class Data : TrackerData() {
+        override fun reset() {
+            cakesOwned.clear()
+        }
+
+        @Expose
+        val cakesOwned: MutableSet<Int> = mutableSetOf()
+
+        @Expose
+        var cakesMissing: MutableSet<Int> = mutableSetOf()
+    }
+
+    private fun addCake(cakeYear: Int) {
+        val storage = storage ?: return
+        if (cakeYear !in storage.cakesOwned) {
+            tracker.modify {
+                it.cakesOwned.add(cakeYear)
+            }
+        }
+        recalculateMissingCakes()
+    }
+
+    private fun removeCake(cakeYear: Int) {
+        val storage = storage ?: return
+        if (cakeYear in storage.cakesOwned) {
+            tracker.modify {
+                it.cakesOwned.remove(cakeYear)
+            }
+        }
+        recalculateMissingCakes()
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && config.enabled
+
+    @HandleEvent
+    fun onCommandRegistration(event: CommandRegistrationEvent) {
+        event.register("shresetcaketracker") {
+            description = "Resets the New Year Cake Tracker"
+            category = CommandCategory.USERS_RESET
+            callback { tracker.resetCommand() }
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (inCakeBag || (inAuctionHouse && (unobtainedCakesDisplayed || searchingForCakes))) {
+            tracker.renderDisplay(config.cakeTrackerPosition, displayModeToggleable = false)
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        val storage = storage
+        if (inAuctionHouse) {
+            unobtainedCakesDisplayed = storage?.let { data ->
+                InventoryUtils.getItemsInOpenChest().onEach { cakeItem ->
+                    cakeNamePattern.matchMatcher(cakeItem.stack.displayName) {
+                        group("year").toInt().takeIf {
```

Comment:
use a cache rather than checking each item each frame. maybe use knownCakesInCurrentInventory? not sure